scripts/xtensa-build-zephyr.py

```diff
@@ -387,16 +388,17 @@ def west_update():
 	execute_command(["west", "update"], check=True, timeout=3000, cwd=west_top)
 
 
-def get_sof_version(abs_build_dir):
-	"""[summary] Get version string major.minor.micro of SOF firmware
-	file. When building multiple platforms from the same SOF commit,
-	all platforms share the same version. So for the 1st platform,
-	generate the version string from sof_version.h and later platforms
-	will reuse it.
+def get_build_and_sof_version(abs_build_dir):
+	"""[summary] Get version string major.minor.micro and build of SOF
+	firmware file. When building multiple platforms from the same SOF
+	commit, all platforms share the same version. So for the 1st platform,
+	generate the version string from sof_version.h and later platforms will
+	reuse it.
 	"""
-	global sof_version
-	if sof_version:
-		return sof_version
+	global sof_fw_version
+	global sof_build_version
```

Comment:
I'm fairly confident you don't need these `global` declarations. Otherwise looks good to me.